x-pack/plugins/xpack_main/server/lib/populate_ui_capabilities.test.ts

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Feature } from './feature_registry';
+import { populateUICapabilities } from './populate_ui_capabilities';
+
+function getMockXpackMainPlugin(features: Feature[]) {
+  return {
+    getFeatures: () => features,
+  };
+}
+
+function getMockOriginalInjectedVars() {
+  return {
+    uiCapabilities: {
+      navLinks: {
+        foo: true,
+        bar: true,
+      },
+      feature: {
+        someCapability: true,
+      },
+      otherFeature: {},
+    },
+  };
+}
+
+function createFeaturePrivilege(key: string, capabilities: string[] = []) {
+  return {
+    [key]: {
+      savedObject: {
+        all: [],
+        read: [],
+      },
+      app: [],
+      ui: [...capabilities],
+    },
+  };
+}
+
+describe('populateUICapabilities', () => {
+  it('handles no original uiCapabilites and no registered features gracefully', () => {
+    const xpackMainPlugin = getMockXpackMainPlugin([]);
+    const originalInjectedVars = {};
+
+    expect(populateUICapabilities(xpackMainPlugin, originalInjectedVars)).toMatchSnapshot();
+  });
+
+  it('returns the original uiCapabilities untouched when no features are registered', () => {
+    const xpackMainPlugin = getMockXpackMainPlugin([]);
+    const originalInjectedVars = getMockOriginalInjectedVars();
+
+    expect(populateUICapabilities(xpackMainPlugin, originalInjectedVars)).toMatchSnapshot();
```

Comment:
To prevent having to jump back and forth between snapshot and test, how do you feel about using `.toMatchInlineSnapshot()`?

Reply:
I almost did that, but I was worried it'd get too large. I'll give it a try! I can always get rid of the snapshot and do object comparisons instead as well